src/ui/components/ShowMoreCard/index.js

```diff
@@ -1,55 +1,122 @@
+/* @flow */
 /* eslint-disable react/no-danger */
-
+import invariant from 'invariant';
 import makeClassName from 'classnames';
 import * as React from 'react';
-import PropTypes from 'prop-types';
 import { compose } from 'redux';
 
 import translate from 'core/i18n/translate';
+import withUIState from 'core/withUIState';
 import { sanitizeHTML } from 'core/utils';
 import Card from 'ui/components/Card';
+import type { I18nType } from 'core/types/i18n';
 
 import './styles.scss';
 
 // This refers to height of `Card-contents` CSS class,
 // beyond which it will add read more link.
 export const MAX_HEIGHT = 150;
 
-export class ShowMoreCardBase extends React.Component {
-  static propTypes = {
-    children: PropTypes.node,
-    className: PropTypes.string,
-    header: PropTypes.node,
-    i18n: PropTypes.object.isRequired,
-  };
+type UIStateType = {|
+  showAllContent: boolean,
+  readMoreExpanded: boolean,
+|};
 
-  constructor(props) {
-    super(props);
-    this.state = { expanded: true };
-  }
+type Props = {|
+  children: React.Element<any> | string,
+  className?: string,
+  header?: React.Element<any> | string,
+  id: string,
+|};
+
+type InternalProps = {|
+  ...Props,
+  i18n: I18nType,
+  setUIState: ($Shape<UIStateType>) => void,
+  uiState: UIStateType,
+  children: React.Element<any>,
+|};
+
+const initialUIState: UIStateType = {
+  showAllContent: true,
+  readMoreExpanded: false,
+};
+
+export class ShowMoreCardBase extends React.Component<InternalProps> {
+  contents: HTMLElement | null;
 
-  componentWillReceiveProps() {
-    this.truncateToMaxHeight(this.contents);
+  componentWillReceiveProps(nextProps: InternalProps) {
+    const { children } = this.props;
+    const { children: newChildren } = nextProps;
+
+    let html =
+      children.props &&
+      children.props.dangerouslySetInnerHTML &&
+      children.props.dangerouslySetInnerHTML.__html;
+
+    let newHtml =
+      newChildren.props &&
+      newChildren.props.dangerouslySetInnerHTML &&
+      newChildren.props.dangerouslySetInnerHTML.__html;
+
+    // If it's not html, let check that text too.
```

Comment:
Let's try to use proper grammar in comments. This should be something like: `If it's not html, check for plain text.`